### PR TITLE
[485607]: HW02 step2

### PR DESCRIPTION
**Pull request by mksoska**

### CODE REVIEW

**Comment by mksoska** [comment]
Done

**Comment by mksoska** [comment]
1805e1707eb7f195ce95552b6af44d884b19c0fa - Finish the entire implementation with newly added tests
9bd7b73b541e0c521854389e6f37dd7aab1e4330 - Fix travis branch build
36f14f1c4daa5ce4697b4be739c43bdb79467f62 - Fix unit test discovery
d25d97fb40cbd04005aec2d57e5320f0c2793d2c - Add one minor test
ef8d88c68682fbb97f4fc7d893535d9a1e3a97d9 - Implement minor additional tests
3155db002c8355fe8eee4655e64629ed6b4700c0 - Finished new implementation
f1b7c3f7726bf986d3c33640b86a3336d875c3ad - Finish old implementation

834516ad0bd8fa98fec4c3009dc531c6160303d5 - Finish whole implementation
d275367d4b17bf342076e26ef14425a9b55c7cfa - Add __sub__ comment
26357a1f9564597662115f1622d01e44880b260a - Add Travis build badge
e48d50348593e60ccfe4ed8c8f432233fa2e6c93 - Do minor README update
b7a67bb9cfd733faa5d700c7361f8943f52f33e5 - Update README with vectors operation
e83e82b36eff835aee614281511eb96b0c638c83 - Update travis branch build
7300266b5f58f0060ec0e7680215dd1ff1947dc4 - Fix travis build, update README with implemented operations
ee49117597d400968ef3945dd208efefef3b014f - Do minor changes
3a23f4b24a97e0ca887575e3f6f2e131dbb0144e - Finish tests implementation
28e9f776e6b9f39c5db998cb911875c09d94cfcb - Begin tests implementation
a48e1c264f241f2ecdc4f4f876c2c7e6c9b46c73 - Finish TODO implementation

**Comment by ph4r05** [comment]
Almost perfect :)

Check travis tests: https://travis-ci.com/crocs-muni/open-source-development-course-hw02-1/jobs/281733838

0 tests run. 

Hints:

- check what .travis.yml does precisely, execute same commands. Are tests executed?
- inspect PR fixing the problem: crocs-muni/open-source-development-course-hw02-2#1

**Review by ph4r05 — CHANGES_REQUESTED** [review]
0 tests run, pls fix

**Comment by mksoska** [comment]
That one was quite tricky :) 
Done the changes.

**Review by ph4r05 — APPROVED** [review]

File: ossdev/__init__.py [hunk 1]
```diff
@@ -21,51 +24,168 @@ def set(self, arr):
...
     def __setitem__(self, key, value):
-        # TODO: implement
-        return None
+        self.d[key] = value
+        return self
 
     def __cmp__(self, other):
-        # TODO: implement, -1 if self < other, 0 if self == other, 1 if self > other
-        return -1
+        if self.length() < other.length():
```
Comment (ph4r05) [inline ossdev/__init__.py:54]: not valid vector comparison. Why only by length?

**Comment by ph4r05** [comment]
Approved, HW done, closing

**Pull request closed without merging**